app/api/events.py

```diff
@@ -492,7 +496,12 @@ def before_update_object(self, event, data, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        if data.get('starts_at') != event.starts_at or data.get('ends_at') != event.ends_at:
+        # deleting event will not trigger validate_date
+        # change of event state from draft to published and also event restoring will trigger validate_date
+        if (data.get('starts_at') and data.get('starts_at') != event.starts_at) or \
+                (data.get('ends_at') and data.get('ends_at') != event.ends_at) or \
+                (data.get('state') == 'published' and event.state == 'draft') or \
+                (event.deleted_at and not data.get('deleted_at')):
```

Comment:
Too much complexity. Most of these checks are present in validate_date. Why is there a need to add so many if else. Simply check if event.state == published or data.state == published

Reply:
I agree. Updated the checks